service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/NotifyGatekeeperController.java

```diff
@@ -51,18 +52,25 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToStartEvent(@RequestBody
         return respond(caseDetails, errors);
     }
 
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmitEvent(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        if (SUBMITTED.equals(caseData.getState())) {
```

Comment:
I'd say this check + test for it isn't needed because event is only available in submitted state, but if we want to make extra sure it's probably fine

Reply:
Isn't a copy of the event available in the gatekeeping state as well

Reply:
yes, tomek mentioned it to me, I forgot about that part - makes sense to keep the test